lib/AST/SearchPathOptions.cpp

```diff
@@ -114,3 +115,40 @@ ModuleSearchPathLookup::searchPathsContainingFile(
                         const ModuleSearchPath *Rhs) { return *Lhs < *Rhs; });
   return Result;
 }
+
+static llvm::Error loadAndValidateVFSOverlay(
+    const std::string &File,
+    const llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> &BaseFS,
+    const llvm::IntrusiveRefCntPtr<llvm::vfs::OverlayFileSystem> &OverlayFS) {
+  auto Buffer = BaseFS->getBufferForFile(File);
+  if (!Buffer)
+    return llvm::createFileError(File, Buffer.getError());
+
+  auto VFS = llvm::vfs::getVFSFromYAML(std::move(Buffer.get()), nullptr, File);
+  if (!VFS)
+    return llvm::createFileError(File, llvm::errc::invalid_argument);
+
+  OverlayFS->pushOverlay(std::move(VFS));
+  return llvm::Error::success();
+}
+
+llvm::Expected<llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>>
+SearchPathOptions::makeOverlayFileSystem(
```

Comment:
It’s fairly self-explaining from the implementation but still, I think it would be good to add a doc comment. Suggestion:
```cpp
/// Create an overlay file system from the \c VFSOverlayFiles set in the \c SearchPathOptions.
```

Reply:
Ah, I missed that, sorry.